tests/evm/test_logs.py

```diff
@@ -0,0 +1,317 @@
+
+    tx = Transaction(id=TX_ID, gas=dynamic_gas)
+
+    steps = [
+        StepState(
+            execution_state=ExecutionState.LOG,
+            rw_counter=1,
+            call_id=CALL_ID,
+            is_root=False,
+            is_create=False,
+            code_source=bytecode_hash,
+            program_counter=0,
+            stack_pointer=1015,
+            memory_size=mstart,
+            gas_left=dynamic_gas,
+            log_id=0,
+            state_write_counter=0,
+        )
+    ]
+    rws = [
+        (1, RW.Read, RWTableTag.Stack, 1, 1015, 0, RLC(mstart, randomness, 8), 0, 0, 0),
+        (2, RW.Read, RWTableTag.Stack, 1, 1016, 0, RLC(msize, randomness, 8), 0, 0, 0),
+        (
+            3,
+            RW.Read,
+            RWTableTag.CallContext,
+            1,
+            CallContextFieldTag.CalleeAddress,
+            0,
+            CALLEE_ADDRESS,
+            0,
+            0,
+            0,
+        ),
+        (
+            4,
+            RW.Write,
+            RWTableTag.TxLog,
+            0,
+            0,
+            TxLogFieldTag.Address,
+            CALLEE_ADDRESS,
+            0,
+            0,
+            0,
+        ),
+        (
+            5,
+            RW.Read,
+            RWTableTag.CallContext,
+            1,
+            CallContextFieldTag.IsStatic,
+            0,
+            0,
+            0,
+            0,
+            0,
+        ),
+    ]
+
+    # append topic rows
+    rws.extend(construct_topic_rws(6, 1017, topics, randomness))
+    new_steps, new_rws = make_log_copy_steps(
+        data,
+        mstart,
+        mstart,
+        msize,
+        rw_counter=rws[-1][0] + 1,
+        program_counter=1,
+        memory_size=next_memory_size,
+        stack_pointer=1015 + (2 + topic_count),
+        gas_left=0,
+        code_source=bytecode_hash,
+        state_write_counter=1,
+        log_id=1,
+    )
+    # append memory & log steps and rows
+    steps.extend(new_steps)
+    rws.extend(new_rws)
+
+    steps.append(
+        StepState(
+            execution_state=ExecutionState.STOP,
+            rw_counter=rws[-1][0] + 1,
+            call_id=CALL_ID,
+            is_root=False,
+            is_create=False,
+            code_source=bytecode_hash,
+            program_counter=1,
+            stack_pointer=1015 + (2 + topic_count),
+            memory_size=next_memory_size,
+            gas_left=0,
+            state_write_counter=1,
+            log_id=1,
+        )
+    )
+
+    tables = Tables(
+        block_table=set(Block().table_assignments(randomness)),
+        tx_table=set(tx.table_assignments(randomness)),
+        bytecode_table=set(bytecode.table_assignments(randomness)),
+        rw_table=set(rws),
+    )
+
+    verify_steps(
+        randomness=randomness,
+        tables=tables,
+        steps=steps,
+    )
+
+
+# helper to construct topics rows of RW table
+def construct_topic_rws(gc: U256, sp: int, topics: list, randomness: int):
```

Comment:
I think you mean `rw_counter` instead of `gc` right?

Reply:
yes, changed to rw_counter in [a9b4e08](https://github.com/appliedzkp/zkevm-specs/pull/125/commits/a9b4e08f53226f01a0310de80b2366a6794129e9)